app/domain/invoices/abacus/membership_invoice.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+module Invoices
+  module Abacus
+    class MembershipInvoice
+
+      MEMBERSHIP_CARD_FIELD_INDEX = 11
+      INVOICE_KIND = :membership
+
+      # person is an Invoices::SacMembership::Person object
+      # role is an membership or new membership Role
+      attr_reader :person, :role
+
+      delegate :context, to: :person
+      delegate :date, :sac, :config, to: :context
+
+      class << self
+
+        # Primarly used for testing purposes.
+        # person is an Invoices::SacMembership::Person object
+        # Role retrieval will usually happen outside of this class.
+        def generate_current(person, client: nil)
+          new(person, current_role(person), client: client).generate
+        end
+
+        def current_role(person)
+          if person.new_entry_role
+            person.new_entry_role
+          elsif person.new_additional_section_membership_roles.present?
+            person.new_additional_section_membership_roles.first
+          else
+            person.main_membership_role
+          end
+        end
+      end
+
+      def initialize(person, role, client: nil)
+        @person = person
+        @role = role
+        @client = client
+      end
+
+      def generate
+        return false unless abacus_person.transmit
+
+        person.model.save
```

Comment:
wieso müssen wir hier `Person#save` aufrufen?

Reply:
In `abacus_person.transmit` wird der `abacus_subject_key` gesetzt.